File: 1-50/problem5/problem5.scala
def gcd(a: Long, b: Long): Long =
  if (a == 0) b
  else gcd(b % a, a)

def lcm(a: Long, b: Long) = a / gcd(a, b) * b

val result = (2 to 20).foldLeft(1L)((a, b) => lcm(a, b))
println(result)

File: 1-50/problem5/problem5.py
def gcd(a, b):
    if a == 0: return b

    while b != 0:
        a, b = b, a % b
    return a

def lcm(a, b):
    return a // gcd(a, b) * b

r = 1
for x in range(2, 20):
    r = lcm(r, x)
print(r)